kube_hunter/modules/hunting/aks.py

```diff
@@ -46,11 +46,16 @@ def get_key_container(self):
             logger.debug("failed getting pod info")
         else:
             pods_data = r.json().get("items", [])
+            suspecious_volume_names = []
             for pod_data in pods_data:
+                for volume in pod_data["spec"]["volumes"]:
```

Comment:
nit: Not sure whether the spec can _not_ have volumes. If that's the case, isn't it worth writing as `for volume in pod_data['spec'].get('volumes',[])` ? Or maybe have a try: except: KeyError?

Reply:
you are absolutely right I forgot to check for failure cases. Thank you for spotting that.